docs/user-documentation.rst

```diff
@@ -0,0 +1,425 @@
+            try:
+                req = Request(environ)
+                resp = req.get_response(self.app)
+                return resp(environ, start_response)
+            finally:
+                self._end_request(req)
+
+        def _end_request(self, req):
+            if req:
+                # Notify Switchboard that the request is finished
+                request_finished.send(req)
+
+
+Using switches
+==============
+
+Once Switchboard is up and running within the application, it's time to begin
+using switches within the code. By default, Switchboard is set to autocreate
+switches, which means that a switch just needs to be checked in code and if
+it doesn't exist it will be created and disabled by default. A switch is always
+referred to by its key, a string name that is expected to be unique to that
+key.
+
+In Python
+---------
+
+To use in Python code (views, models, etc.), import the operator singleton
+and use the ``is_active`` method to see if the switch is on or not::
+
+    from switchboard import operator
+    ...
+    if operator.is_active('foo'):
+        ... do something ...
+    else:
+        ... do something else ...
+
+If autocreate is on (and it is by default), the 'foo' switch will be
+automatically created and set to disabled the first time it is referenced.
+Activating the switch and controlling exactly when the switch is active,
+are covered in `Managing switches`_.
+
+In templates
+------------
+
+Every templating framework has its own take on how (or even if) logic may be
+used. That said, Switchboard provides some helpers to make things easier. If
+
+In javascript
+-------------
+
+The easiest way to use Switchboard in conjunction with Javascript is to set a
+Javascript flag within your template code and then use that flag within your
+Javascript logic. Using Mako's syntax in the template::
+
+    <%!
+        from switchboard import operator
+    %>
+    <script>
+        window.switches = window.switches || {};
+        % if operator.is_active('foo'):
+        switches.foo = true;
+        % else:
+        switches.foo = false;
+        % endif
+    </script>
+
+In the Javascript::
+
+    if (switches.foo) {
+        ... do something ...
+    } else {
+        ... do something else ...
+    }
+
+Again, this time using Jinja syntax and the Switchboard-provided "active"
+test_::
+
+    <script>
+        window.switches = {};
+        switches.foo = {{ 'true' if 'foo' is active else 'false' }};
+    </script>
+
+Context objects
+---------------
+
+Every switch is evaluated (to see if it is active or not) within a particular
+context. By default, that context includes the request object, which allows
+Switchboard to specify conditions such as: "make this switch active only for
+requests with 'foo' in the query string." That said, there may be other
+objects that would be handy to have available in the context. For example, in
+an ecommerce setting, the Product model may have a ``new`` flag. By passing
```

Comment:
Google returns 127M hits for 'ecommerce' but a whopping 354M hits for 'e-commerce'. I like the hyphenated version as it is clear we're using jargon.